### PR TITLE
HTCONDOR-2509 Document the new -hip option to condor_gpu_discovery.

### PR DESCRIPTION
**Pull request by johnkn**
Also allow both -cuda and -hip options to be used at the same time to
  mean that just Cuda and HIP detection (but not OpenCL) is desired.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/man-pages/condor_gpu_discovery.rst [hunk 1]
```diff
@@ -15,18 +15,31 @@ Description
...
+This tool is not available for MAC OS platforms.
+
+The ``-cuda``, ``-hip`` and ``-opencl`` arguments control which libraries
+are used for discovering GPUs. If any of these libraries report GPUs,
+then detection will stop at that library unless more then one
+of the above command line arguments is used. If none if these arguments is used,
+*condor_gpu_discovery* will first try to detect GPU devices using
+the NVidia driver library, then the HIP 6 library, and
+finally the OpenCL library. Because OpenCL devices do not have unique
+identifiers, If the ``-opencl`` argument is
```
Comment (Todd-L-Miller) [inline docs/man-pages/condor_gpu_discovery.rst:34]:
```suggestion
identifiers, if the ``-opencl`` argument is
```

File: docs/man-pages/condor_gpu_discovery.rst [hunk 2]
```diff
@@ -15,18 +15,31 @@ Description
...
+be reported only if they are the only type of device reported.
+
 If ``CUDA_VISIBLE_DEVICES`` or ``GPU_DEVICE_ORDINAL`` is set in the
 environment when *condor_gpu_discovery* is run, it will report only
-devices present in the those lists.
-
-This tool is not available for MAC OS platforms.
+NVidia devices present in the those lists.  If ``HIP_VISIBLE_DEVICES``
+or ``ROCR_VISIBLE_DEVICES`` is set in the environment, it will report
+only HIP devices present on those lists.
```
Comment (Todd-L-Miller) [inline docs/man-pages/condor_gpu_discovery.rst:42]: Which list wins?  Or is it the intersection?

File: src/gpu/condor_gpu_discovery.cpp [hunk 3]
```diff
@@ -590,6 +592,8 @@ main( int argc, const char** argv)
 	#if defined(WINDOWS)
 		_putenv( "CUDA_VISIBLE_DEVICES=" );
 		_putenv( "GPU_DEVICE_ORDINAL=" );
+		_putenv( "HIP_VISIBLE_DEVICES=" );
+		_putenv( "ROCR_VISIBLE_DEVICES=" );
```
Comment (Todd-L-Miller) [inline src/gpu/condor_gpu_discovery.cpp:596]: These environment variables don't exist on other platforms?

Comment (johnkn) [reply src/gpu/condor_gpu_discovery.cpp:596]: they were already being cleared on non-windows.

File: docs/man-pages/condor_gpu_discovery.rst (hunk 2 above)
Comment (johnkn) [reply docs/man-pages/condor_gpu_discovery.rst:42]: union

**Review by Todd-L-Miller — APPROVED** [review]
LGTM.

File: docs/man-pages/condor_gpu_discovery.rst (hunk 1 above)
Comment (Todd-L-Miller) [reply docs/man-pages/condor_gpu_discovery.rst]: Fixed.

**Pull request merged**